torch/nn/modules/activation.py

```diff
@@ -1062,6 +1061,11 @@ def forward(self, query: Tensor, key: Tensor, value: Tensor, key_padding_mask: O
             `batch_first` argument is ignored for unbatched inputs.
         """
         is_batched = query.dim() == 3
+        if key_padding_mask is not None:
+            if key_padding_mask.dtype != torch.bool:
+                if not torch.is_floating_point(key_padding_mask):
+                    raise AssertionError(
+                        "only bool and floating type of key_padding_mask is supported")
```

Comment:
You are adding the same check thrice, which seems to warrant a convenience function
```
def _assert_if_tensor_is_not_float_or_bool(t: Optional[torch.Tensor], name: str) -> None:
    if t is None:
        return
   if t.dtype != torch.bool and not t.dtype.is_floating_point:
       raise AssertionError(f"only bool and floating types of {name} are supported")
```

Reply:
That is a good idea, but dont want to add new helper functions in this diff anymore in case it will break torch script :(